Broken package: libsodium + Visual Studio 2017

Package `libsodium` does not work with Visual Studio 2017. I followed the instructions and opened an issue. I also mention @Cyberunner23 as suggested.

Below is the generated error:
```
[hunter] Initializing Hunter workspace (2c416fea5f66429ffb4412ce5b8d0c6799ec44d2)
[hunter]   https://github.com/ruslo/hunter/archive/v0.19.109.tar.gz
[hunter]   -> C:/.hunter/_Base/Download/Hunter/0.19.109/2c416fe
[hunter] Calculating Toolchain-SHA1
[hunter] Calculating Config-SHA1
[hunter] HUNTER_ROOT: C:/.hunter
[hunter] [ Hunter-ID: 2c416fe | Toolchain-ID: 2d5f6bd | Config-ID: f9659d0 ]
[hunter] LIBSODIUM_ROOT: C:/.hunter/_Base/2c416fe/2d5f6bd/f9659d0/Install (ver.: 1.0.10)
[hunter] Building libsodium
loading initial cache file C:/.hunter/_Base/2c416fe/2d5f6bd/f9659d0/cache.cmake

loading initial cache file C:/.hunter/_Base/2c416fe/2d5f6bd/f9659d0/Build/libsodium/args.cmake

-- The C compiler identification is MSVC 19.11.25508.2

-- The CXX compiler identification is MSVC 19.11.25508.2

-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.11.25503/bin/HostX86/x86/cl.exe

-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.11.25503/bin/HostX86/x86/cl.exe -- works

-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.11.25503/bin/HostX86/x86/cl.exe

-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.11.25503/bin/HostX86/x86/cl.exe -- works

-- Detecting CXX compile features

-- Detecting CXX compile features - done



[hunter ** FATAL ERROR **] Visual studio version not supported. Supported versions are: Visual studio 2010, 2012, 2013 and 2015.
[hunter ** FATAL ERROR **] If there is a new version please open an issue at https://github.com/ruslo/hunter and mention @Cyberunner23
[hunter ** FATAL ERROR **] [Directory:C:/.hunter/_Base/2c416fe/2d5f6bd/f9659d0/Build/libsodium]


------------------------------ WIKI -------------------------------

    https://github.com/ruslo/hunter/wiki/error.incorrect.input.data
-------------------------------------------------------------------


CMake Error at C:/.hunter/_Base/Download/Hunter/0.19.109/2c416fe/Unpacked/cmake/modules/hunter_wiki.cmake:12 (message):

Call Stack (most recent call first):
  C:/.hunter/_Base/Download/Hunter/0.19.109/2c416fe/Unpacked/cmake/modules/hunter_fatal_error.cmake:20 (hunter_wiki)
  C:/.hunter/_Base/Download/Hunter/0.19.109/2c416fe/Unpacked/cmake/modules/hunter_user_error.cmake:7 (hunter_fatal_error)
  CMakeLists.txt:48 (hunter_user_error)



-- Configuring incomplete, errors occurred!

See also "C:/.hunter/_Base/2c416fe/2d5f6bd/f9659d0/Build/libsodium/Build/CMakeFiles/CMakeOutput.log".


[hunter ** FATAL ERROR **] Configure step failed (dir: C:/.hunter/_Base/2c416fe/2d5f6bd/f9659d0/Build/libsodium)
[hunter ** FATAL ERROR **] [Directory:C:/.hunter/_Base/Download/Hunter/0.19.109/2c416fe/Unpacked/cmake/projects/libsodium]

------------------------------ WIKI -------------------------------
    https://github.com/ruslo/hunter/wiki/error.external.build.failed
-------------------------------------------------------------------

CMake Error at C:/.hunter/_Base/Download/Hunter/0.19.109/2c416fe/Unpacked/cmake/modules/hunter_wiki.cmake:12 (message):
Call Stack (most recent call first):
  C:/.hunter/_Base/Download/Hunter/0.19.109/2c416fe/Unpacked/cmake/modules/hunter_fatal_error.cmake:20 (hunter_wiki)
  C:/.hunter/_Base/Download/Hunter/0.19.109/2c416fe/Unpacked/cmake/modules/hunter_download.cmake:570 (hunter_fatal_error)
  C:/.hunter/_Base/Download/Hunter/0.19.109/2c416fe/Unpacked/cmake/projects/libsodium/hunter.cmake:28 (hunter_download)
  C:/.hunter/_Base/Download/Hunter/0.19.109/2c416fe/Unpacked/cmake/modules/hunter_add_package.cmake:53 (include)
  cpp/sodium-backend/CMakeLists.txt:1 (hunter_add_package)


Configuring incomplete, errors occurred!
See also "C:/Users/emile/Desktop/ssp21-cpp/build/CMakeFiles/CMakeOutput.log".
```

--- a/cmake/projects/libsodium/schemes/url_sha1_libsodium_msbuild.cmake.in
+++ b/cmake/projects/libsodium/schemes/url_sha1_libsodium_msbuild.cmake.in
@@ -35,6 +35,7 @@ string(COMPARE EQUAL "@HUNTER_MSVC_VERSION@" "10" _is_platform_toolset_v100) # V
 string(COMPARE EQUAL "@HUNTER_MSVC_VERSION@" "11" _is_platform_toolset_v110) # VC2012
 string(COMPARE EQUAL "@HUNTER_MSVC_VERSION@" "12" _is_platform_toolset_v120) # VC2013
 string(COMPARE EQUAL "@HUNTER_MSVC_VERSION@" "14" _is_platform_toolset_v140) # VC2015
+string(COMPARE EQUAL "@HUNTER_MSVC_VERSION@" "15" _is_platform_toolset_v141) # VC2017
 
 if(_is_platform_toolset_v100)
     set(msvc_platform_toolset "v100")
@@ -44,9 +45,11 @@ elseif(_is_platform_toolset_v120)
     set(msvc_platform_toolset "v120")
 elseif(_is_platform_toolset_v140)
     set(msvc_platform_toolset "v140")
+elseif(_is_platform_toolset_v141)
+    set(msvc_platform_toolset "v141")
 else()
     hunter_user_error(
-        "Visual studio version not supported. Supported versions are: Visual studio 2010, 2012, 2013 and 2015."
+        "Visual studio version not supported. Supported versions are: Visual studio 2010, 2012, 2013, 2015 and 2017."
         "If there is a new version please open an issue at https://github.com/ruslo/hunter and mention @Cyberunner23")
 endif()
 

